src/components/AssignRoleModal/AssignRoleModal.tsx

```diff
@@ -1,16 +1,19 @@
 import { Button } from '@/components/Button/Button'
 
-import styles from './AssignRole.module.scss'
 import modalStyles from '@/components/Modal/Modal.module.scss'
+import { Role } from '@prisma/client'
+import styles from './AssignRole.module.scss'
 
 export default function AssignRoleModal({
   profileId,
+  userRoles,
   onClose,
 }: {
   profileId: string
+  userRoles: Role[]
   onClose: () => void
 }) {
-  return (
+  return !userRoles.includes(Role.MODERATOR) ? (
```

Comment:
**nitpick:** I think you don't have to pass user roles because you can pull it out straight in component using `useSession` hook